includes/settings/class-wcpdf-settings-debug.php

```diff
@@ -520,10 +519,45 @@ public function ajax_process_danger_zone_tools() {
 			'paginate'       => true,
 			'posts_per_page' => 50,
 			'page'           => $page_count,
-			$date_type       => $from_date . '...' . $to_date,
 		);
-	
-		$results   = wc_get_orders( $args );
+		
+		$date_type     = isset( $request['date_type'] ) ? esc_attr( $request['date_type'] ) : '';
+		$wc_date_types = array(
+			'date_created',
+			'date_modified',
+			'date_completed',
+			'date_paid',
+		);
+		
+		if ( in_array( $date_type, $wc_date_types ) ) {
+			$date_arg      = $date_type;
+		} elseif ( 'document_date' === $date_type ) {
+			$document_slug = str_replace( '-', '_', $document_type );
+			$date_arg      = "wcpdf_{$document_slug}_date";
+		} else {
+			$date_arg      = '';
+		}
+		
+		if ( empty( $date_arg ) ) {
+			$message = __( 'Wrong date type selected.', 'woocommerce-pdf-invoices-packing-slips' );
+			wp_send_json_error( compact( 'message' ) );
+		}
+		
+		$args[ $date_arg ] = $from_date . '...' . $to_date;
+		
+		if ( WPO_WCPDF()->order_util->custom_orders_table_usage_is_enabled() ) {
+			$args = wpo_wcpdf_parse_document_date_for_wp_query( $args, $args );
+		} else {
+			add_filter( 'woocommerce_order_data_store_cpt_get_orders_query', 'wpo_wcpdf_parse_document_date_for_wp_query', 10, 2 );
```

Comment:
I think we can directly call that method for both types and remove this filter here.

Reply:
The `meta_query` was introduced only on `6.9.0` I believe ( I thought it was with HPOS), so I think it will not work on older versions of Woo.

Reply:
Isn't that PR for before releasing the HPOS?
Also, I tested with WooCommerce 6.8.0, and the `wpo_wcpdf_parse_document_date_for_wp_query` was working correctly.

Reply:
Fixed.